File: Python/Garage.py
# Saya Muhammad Rizki Revandi dengan NIM 2205027 mengerjakan Latihan Praktikum 4
# dalam mata kuliah Desain Pemrograman Berorientasi Objek untuk keberkahanNya maka saya tidak
# melakukan kecurangan seperti yang telah dispesifikasikan. Aamiin.

class Garage:
    __namaGarasi = ""
    __luasGarasi = 0
    __daftarKendaraan = []
    
    def __init__(self, namaGarasi, luasGarasi):
        self.__namaGarasi = namaGarasi
        self.__luasGarasi = luasGarasi
        self.__daftarKendaraan = []
    
    def getNamaGarasi(self):
        return self.__namaGarasi
    
    def setNamaGarasi(self, namaGarasi):
        self.__namaGarasi = namaGarasi
        
    def getLuasGarasi(self):
        return self.__luasGarasi
    
    def setLuasGarasi(self, luasGarasi):
        self.__luasGarasi = luasGarasi
        
    def getDaftarKendaraan(self):
        return self.__daftarKendaraan
        
    def addDaftarKendaraan(self, daftarKendaraan):
        self.__daftarKendaraan.append(daftarKendaraan)

File: CPP/Garage.cpp
#include <iostream>
#include <vector>
#include <string>

using namespace std;

class Garage
{
private:
    string namaGarasi;
    int luasGarasi;
    vector<Vehicle*> daftarKendaraan;

public:
    Garage()
    {
    }

    Garage(string namaGarasi, int luasGarasi)
    {
        this->namaGarasi = namaGarasi;
        this->luasGarasi = luasGarasi;
    }

    string getNamaGarasi()
    {
        return this->namaGarasi;
    }

    void setNamaGarasi(string namaGarasi)
    {
        this->namaGarasi = namaGarasi;
    }

    int getLuasGarasi()
    {
        return this->luasGarasi;
    }

    void setLuasGarasi(int luasGarasi)
    {
        this->luasGarasi = luasGarasi;
    }

    vector<Vehicle*> getDaftarKendaraan()
    {
        return daftarKendaraan;
    }

    void addDaftarKendaraan(Vehicle* kendaraan)
    {
        daftarKendaraan.push_back(kendaraan);
    }

    ~Garage()
    {
    }
};
